vm/src/stdlib/socket.rs

```diff
@@ -449,6 +449,28 @@ fn socket_htonl(host: u32, vm: &VirtualMachine) -> PyResult {
     Ok(vm.new_int(host.to_be()))
 }
 
+fn socket_htons(host: u16, vm: &VirtualMachine) -> PyResult {
```

Comment:
if the only difference is type, probably generic function will work
```rust
fn socket_hton<U>(host: U, _vm: &VirtualMachine) -> U {
    host.to_be()
}
```

and make rustfunc like this:
```rust
"htons" => ctx.new_rustfunc(socket_hton::<u16>),
```

Reply:
I think you'd have to give the trait bound `U: num_traits::PrimInt` for that to work.